evm_arithmetization/src/prover.rs

```diff
@@ -560,6 +576,23 @@ pub fn check_abort_signal(abort_signal: Option<Arc<AtomicBool>>) -> Result<()> {
     Ok(())
 }
 
+/// Returns a vector containing the data required to generate all segments for
+/// one full execution.
+pub fn generate_all_data_segments<F: RichField>(
+    max_cpu_len_log: Option<usize>,
+    inputs: GenerationInputs,
+) -> anyhow::Result<Vec<GenerationSegmentData>> {
+    // TODO: change this value after some benchmarks have been carried out?
```

Comment:
I don't think we can hardcode a nice value here, or it'll have to be an approximative one. There isn't a good metric that can give us right away the number of segments, gas cost highly depends on the table it involves, and CPU cycles need to be executed first.
Just saying, I guess it's fine if we have a somewhat arbitrary value here (though I'd prefer something rounded like 100 😛)

Reply:
Indeed, it would have to be an extremely approximative value (or even arbitrary)... I just thought that given it's a `Vec` to which we need to append, maybe there was a way to reduce the overhead? But I guess there might also be cases with so many segments that it might make things much worse, so I might as well remove it...